### PR TITLE
Remove splitting ranges on emojis on the web

### PR DESCRIPTION
**Pull request by Skalakid**
### Details

This PR fixes the problem with triplicating syntaxes inside the Live Markdown Input, by removing range-splitting logic from the web platform and replacing it with the fix based on CSS styles. In the future, we will focus on fixing the web parser to support splitting emojis among all other styles (especially inline code block) and properly handle tag hierarchy edge cases when building HTML structure. 

### Related Issues

https://github.com/Expensify/App/issues/55115

### Manual Tests

1. To the text that contains emojis, add italics styling, e.g. `_🥰 test 😇 test 🥹_`
2. Verify if the text has italics style and emojis don't
3. Verify if it still works when adding other styles around it, e.g. `# *~_🥰 test 😇 test 🥹_~*`
4. Verify if emojis aren't strikethrough
5. Test it with different style combinations

### Linked PRs

https://github.com/Expensify/react-native-live-markdown/pull/597

### CODE REVIEW

File: example/src/testConstants.ts [hunk 1]
```diff
@@ -1,18 +1,6 @@
...
-  '# header1',
-  '> blockquote',
-  '`inline code`',
-  '```\ncodeblock\n```',
-  '@here',
-  '@someone@swmansion.com',
-  '#mention-report',
-  '![demo image](https://picsum.photos/id/1067/200/300)',
-].join('\n');
+const EXAMPLE_CONTENT = ['~_😀🍕🍔_~'].join('\n');
```
Comment (BartoszGrajdek) [inline example/src/testConstants.ts:3]: Can you revert that part?

File: src/parseExpensiMark.ts [hunk 2]
```diff
@@ -252,11 +252,14 @@ function parseExpensiMark(markdown: string): MarkdownRange[] {
...
-
-  let splittedRanges = splitRangesOnEmojis(ranges, 'italic');
-  splittedRanges = splitRangesOnEmojis(splittedRanges, 'strikethrough');
-
-  const sortedRanges = sortRanges(splittedRanges);
+  let markdownRanges = ranges;
+  // Blocks applying italic and strikethrough styles to emojis on Android
+  // TODO: Remove this condition when splitting emojis inside the inline code block will be fixed on the web
+  if (Platform.OS === 'android') {
+    markdownRanges = splitRangesOnEmojis(markdownRanges, 'italic');
```
[the remark this answers is not in the document]
Comment (Skalakid) [reply src/parseExpensiMark.ts:260]: Changed in [2fe0e40](https://github.com/Expensify/react-native-live-markdown/pull/598/commits/2fe0e40c7f8e62538ef8c44cb52d7e297f17c026)

**Review by BartoszGrajdek — DISMISSED** [review]
LGTM! 👍🏻 

One problem I have is that it's not possible to use `CMD + ➡️` when the cursor is in front of emoji. (not sure if that's a blocker cc @tomekzaw)

<details>
<summary>SR</summary>


https://github.com/user-attachments/assets/79ab6301-20cd-441f-b57e-993e8cf6d0c0



</details>

File: src/parseExpensiMark.ts [hunk 3]
```diff
@@ -252,13 +252,16 @@ function parseExpensiMark(markdown: string): MarkdownRange[] {
...
+    markdownRanges = splitRangesOnEmojis(markdownRanges, 'strikethrough');
+  }
 
-  let splittedRanges = splitRangesOnEmojis(ranges, 'italic');
-  splittedRanges = splitRangesOnEmojis(splittedRanges, 'strikethrough');
-
-  const sortedRanges = sortRanges(splittedRanges);
-  const groupedRanges = groupRanges(sortedRanges);
-
+  const groupedRanges = groupRanges(markdownRanges);
```
[the remark this answers is not in the document]
Comment (Skalakid) [reply src/parseExpensiMark.ts:264]: We don't need it here. Sorting influences only the web platform and since we blocked spitting ranges for this platform, the output will still be correctly sorted

File: src/parseExpensiMark.ts [hunk 4]
```diff
@@ -252,13 +252,16 @@ function parseExpensiMark(markdown: string): MarkdownRange[] {
     );
     return [];
   }
+  const sortedRanges = sortRanges(ranges);
+  let markdownRanges = sortedRanges;
+  if (Platform.OS === 'android') {
```
Comment (tomekzaw) [inline src/parseExpensiMark.ts:257]: Since `parseExpensiMark` is a worklet, let's extract `Platform.OS` outside to avoid capturing whole `Platform` object in worklet closure.
```ts
const isAndroid = Platform.OS === 'android';

function parseExpensiMark(...) {
  if (isAndroid) { /* ... */ }
}
```

File: src/parseExpensiMark.ts [hunk 5]
```diff
@@ -252,13 +252,16 @@ function parseExpensiMark(markdown: string): MarkdownRange[] {
     );
     return [];
   }
+  const sortedRanges = sortRanges(ranges);
+  let markdownRanges = sortedRanges;
```
Comment (tomekzaw) [inline src/parseExpensiMark.ts:256]:
```suggestion
  let markdownRanges = sortRanges(ranges);
```

File: src/web/utils/blockUtils.ts [hunk 6]
```diff
@@ -26,7 +26,13 @@ function addStyleToBlock(targetElement: HTMLElement, type: NodeType, markdownSty
       node.style.textDecoration = 'line-through';
       break;
     case 'emoji':
-      Object.assign(node.style, {...markdownStyle.emoji, verticalAlign: 'middle'});
+      Object.assign(node.style, {
+        ...markdownStyle.emoji,
+        verticalAlign: 'middle',
+        fontStyle: 'normal',
+        textDecoration: 'none',
```
Comment (tomekzaw) [inline src/web/utils/blockUtils.ts:33]:
```suggestion
        fontStyle: 'normal', // remove italic
        textDecoration: 'none', // remove strikethrough
```

**Review by tomekzaw — APPROVED** [review]
